marl_eval/plotting_tools/plotting.py

```diff
@@ -312,6 +347,14 @@ def sample_efficiency_curves(
     data_dictionary = upper_algo_dict
     algorithms = list(data_dictionary.keys())
 
+    if legend_map is not None:
+        legend_map = {algo.upper(): value for algo, value in legend_map.items()}
+        # Replace keys in data dict with corresponding key in legend map
+        data_dictionary = {
```

Comment:
maybe it will be better if we have assert here that if an error happens we print the list of algorithms that exist in the json file, to make the user check which of the list of algorithms he is missing or misspelled

Reply:
We could do this, but by default it will already complain if the keys in the legend map do not line up with the algorithms from the dataset. Happy to still make an assert here if you think it would be more useful.

Reply:
Yeah I tested it out and find it will show the wrong key, so we can skip the assert I guess